### PR TITLE
Improve nncvxpro notebook

### PR DESCRIPTION
**Pull request by mathurinm**
- fixed the step in ISTA (there was no square in ||A||), also fixed the typo in the Markdown formula
- used sklearn.preprocessing.StandardScaler() to scale `A`, otherwise since some pixels are constants across all images you obtain nan when dividing by A.std()
- used scipy.optimize.check_grad() to check the gradient of `f`
- used from numpy.linarlf import norm`, this is canon to have shorter lines in formulas
- removed the use of `np.inner`, used `@` instead.
- used `plt.semilogy` instead of `plt.plot(..., np.log10(...))`
- f-strings to format result: `print(f"variable my_var is equal to {my_var}") 
- use `np;arange()` to obtain integers instead of `np.linspace().astype(int)`
- used `A.toarray()` to convert a sparse matrix directly to a numpy array
- seeded at the beginning of the nb to guarantee deterministic execution 
- also some PEP8 coding style (`f(a)` not `f( a )`)

@gpeyre

### CODE REVIEW

**Comment by mathurinm** [comment]
I do not get the same figure as when I downloaded the notebook (celer is faster on my side) so you may checkout the PR locally and change the dataset or decrease lambda
